Objects/setobject.c

```diff
@@ -1277,6 +1474,14 @@ set_swap_bodies(PySetObject *a, PySetObject *b)
         FT_ATOMIC_STORE_SSIZE_RELAXED(a->hash, -1);
         FT_ATOMIC_STORE_SSIZE_RELAXED(b->hash, -1);
     }
+    if (!SET_IS_SHARED(b) && SET_IS_SHARED(a)) {
```

Comment:
`set_swap_bodies` is only called in two locations in this file. For both cases the second argument `b` is a newly created temporary which is discarded afterwards. So we could replace part (all?) of these branches with asserts.  Part of the other work (e.g. copying back to `b` not be needed for the same reason.